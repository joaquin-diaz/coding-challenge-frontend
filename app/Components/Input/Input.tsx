import * as React from 'react';
import styled from '@emotion/styled';

const StyledInput = styled.input`
  border: none;
  width: 100%;
  font-size: 14px;

  &:hover,
  &:focus {
    outline: none;
  }
`;

interface InputProps {
  value: string;
  onChange: (value: string) => void;
  placeholder: string;
}

const Input: React.FC<InputProps> = props => {
  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (event && event.target) {
      props.onChange(event.target.value);
    }
  };

  return <StyledInput value={props.value} onChange={onChange} placeholder={props.placeholder} />;
};

export default Input;
