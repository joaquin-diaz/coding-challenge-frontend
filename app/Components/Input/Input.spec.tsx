import * as React from 'react';

import Input from './Input';
import { mount } from 'enzyme';

const noop = () => {};

describe('<Input />', () => {
  it('should render', () => {
    const wrapper = mount(<Input onChange={noop} placeholder="Test" value="123" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should return the value on change', () => {
    const mockOnChange = jest.fn();
    const wrapper = mount(<Input onChange={mockOnChange} placeholder="Test" value="123" />);

    wrapper.find('input').simulate('change', { target: { value: 'asd123' } });

    expect(mockOnChange).toHaveBeenCalledWith('asd123');
  });
});
