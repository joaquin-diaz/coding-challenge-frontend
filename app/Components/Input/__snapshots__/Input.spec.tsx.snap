// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render 1`] = `
.emotion-0 {
  border: none;
  width: 100%;
  font-size: 14px;
}

.emotion-0:hover,
.emotion-0:focus {
  outline: none;
}

<Input
  onChange={[Function]}
  placeholder="Test"
  value="123"
>
  <Styled(input)
    onChange={[Function]}
    placeholder="Test"
    value="123"
  >
    <input
      className="emotion-0"
      onChange={[Function]}
      placeholder="Test"
      value="123"
    />
  </Styled(input)>
</Input>
`;
